pkg/operation/botanist/namespaces_test.go

```diff
@@ -150,11 +190,93 @@ var _ = Describe("Namespaces", func() {
+				kubernetesClient.EXPECT().Client().Return(c),
+				c.EXPECT().List(ctx, gomock.AssignableToTypeOf(&gardencorev1beta1.ControllerRegistrationList{})).DoAndReturn(func(_ context.Context, list client.ObjectList, _ ...client.ListOption) error {
+					(controllerRegistrationList).DeepCopyInto(list.(*gardencorev1beta1.ControllerRegistrationList))
+					return nil
+				}),
+				c.EXPECT().Create(ctx, obj),
+			)
+
+			Expect(botanist.DeploySeedNamespace(ctx)).To(Succeed())
+			Expect(botanist.SeedNamespaceObject).To(Equal(obj))
+		})
+
+		It("should successfully remove extension labels from the namespace when extensions are deleted from shoot spec or marked as disabled", func() {
+			botanist.Shoot.Info.Spec.Extensions = []gardencorev1beta1.Extension{
+				{
+					Type: extensionType1,
+				},
+				{
+					Type: extensionType2,
+				},
+			}
+			obj.Labels[v1beta1constants.LabelExtensionPrefix+extensionType1] = "true"
+			obj.Labels[v1beta1constants.LabelExtensionPrefix+extensionType2] = "true"
+			obj.Labels[v1beta1constants.LabelExtensionPrefix+extensionType3] = "true"
+
+			gomock.InOrder(
+				kubernetesClient.EXPECT().Client().Return(c),
+				c.EXPECT().Get(ctx, kutil.Key(namespace), gomock.AssignableToTypeOf(&corev1.Namespace{})).Return(apierrors.NewNotFound(schema.GroupResource{}, "")),
+				kubernetesClient.EXPECT().Client().Return(c),
+				c.EXPECT().List(ctx, gomock.AssignableToTypeOf(&gardencorev1beta1.ControllerRegistrationList{})).DoAndReturn(func(_ context.Context, list client.ObjectList, _ ...client.ListOption) error {
+					(controllerRegistrationList).DeepCopyInto(list.(*gardencorev1beta1.ControllerRegistrationList))
+					return nil
+				}),
+				c.EXPECT().Create(ctx, obj),
+				kubernetesClient.EXPECT().Client().Return(c),
+				c.EXPECT().Get(ctx, kutil.Key(namespace), gomock.AssignableToTypeOf(&corev1.Namespace{})).Return(apierrors.NewNotFound(schema.GroupResource{}, "")),
+				kubernetesClient.EXPECT().Client().Return(c),
+				c.EXPECT().List(ctx, gomock.AssignableToTypeOf(&gardencorev1beta1.ControllerRegistrationList{})).DoAndReturn(func(_ context.Context, list client.ObjectList, _ ...client.ListOption) error {
+					(controllerRegistrationList).DeepCopyInto(list.(*gardencorev1beta1.ControllerRegistrationList))
+					return nil
+				}),
 				c.EXPECT().Create(ctx, obj),
 			)
 
 			Expect(botanist.DeploySeedNamespace(ctx)).To(Succeed())
 			Expect(botanist.SeedNamespaceObject).To(Equal(obj))
+
+			botanist.Shoot.Info.Spec.Extensions = []gardencorev1beta1.Extension{
+				{
+					Type: extensionType1,
+				},
+				{
+					Type:     extensionType3,
+					Disabled: pointer.Bool(true),
+				},
+			}
+
+			delete(obj.Labels, v1beta1constants.LabelExtensionPrefix+extensionType2)
+			delete(obj.Labels, v1beta1constants.LabelExtensionPrefix+extensionType3)
+
+			Expect(botanist.DeploySeedNamespace(ctx)).To(Succeed())
+			Expect(botanist.SeedNamespaceObject).To(Equal(obj))
```

Comment:
Instead of calling `botanist.DeploySeedNamespace` 2 times, you can adapt the test configuration to return a preconfigured Namespace (instead of `not found` err) when seed client is called to get the Namespace (L217).